### PR TITLE
Build with dotnet

### PR DESCRIPTION
**Pull request by victorhurdugaci**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,9 +10,11 @@ addons:
     - libssl-dev
     - libunwind8
     - zlib1g
-env:
-  - KOREBUILD_DNU_RESTORE_CORECLR=true KOREBUILD_TEST_DNXCORE=true
 mono:
   - 4.0.5
+os:
+  - linux
+  - osx
```
Comment (dougbu) [inline .travis.yml]: Should include install step for OSx that does `brew install icu4c`. See aspnet/Universe/issues/349.

File: src/Microsoft.AspNet.Testing/project.json [hunk 2]
```diff
@@ -10,8 +10,7 @@
     "url": "git://github.com/aspnet/testing"
```
Comment (dougbu) [inline src/Microsoft.AspNet.Testing/project.json]: Why no `testRunner` setting here?

File: src/Microsoft.AspNet.Testing/project.json [hunk 3]
```diff
@@ -19,8 +18,15 @@
         "System.Runtime": "",
         "System.Reflection": "",
         "System.Threading.Tasks": ""
+      },
+      "dependencies": {
+        "xunit": "2.1.0"
```
Comment (dougbu) [inline src/Microsoft.AspNet.Testing/project.json]: `xunit.runner.console`

File: src/Microsoft.AspNet.Testing/project.json [hunk 4]
```diff
@@ -10,8 +10,7 @@
     "url": "git://github.com/aspnet/testing"
   },
   "dependencies": {
-    "Microsoft.Extensions.PlatformAbstractions": "1.0.0-*",
-    "xunit.runner.aspnet": "2.0.0-aspnet-*"
+    "Microsoft.Extensions.PlatformAbstractions": "1.0.0-*"
```
Comment (dougbu) [inline src/Microsoft.AspNet.Testing/project.json]: Why no `xunit` here?

File: test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs [hunk 5]
```diff
@@ -20,13 +20,11 @@ public class ProjectTest
...
-            var libraryManager = PlatformServices.Default.LibraryManager;
-
-            // Example: "C:\github\testing\test\Sample.Tests"
-            _testProjectPath = Path.GetDirectoryName(libraryManager.GetLibrary(projectName).Path);
+            var testProjectFolder = Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(Directory.GetCurrentDirectory())));
+            _testProjectPath = Path.Combine(testProjectFolder, projectName);
         }
 
-        [Fact]
+        [Fact(Skip = "Doesn't run on dotnet")]
```
Comment (dougbu) [inline test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs]: This doesn't tell us anything. _Why_ doesn't this run on `dotnet`? Could the test have conditions that allow it to run on at least a few system / platform combinations/

File: test/Microsoft.Dnx.TestHost.Tests/TestHostTest.cs [hunk 6]
```diff
@@ -21,11 +21,11 @@ public class TestHostTest
...
-            var libraryManager = PlatformServices.Default.LibraryManager;
-            _testProject = Path.GetDirectoryName(libraryManager.GetLibrary("Sample.Tests").Path);
+            var testProjectFolder = Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(Directory.GetCurrentDirectory())));
+            _testProject = Path.Combine(testProjectFolder, "Sample.Tests");
         }
 
-        [ConditionalFact]
+        [ConditionalFact(Skip = "Doesn't support dotnet")]
         [OSSkipCondition(OperatingSystems.Linux)]
         [OSSkipCondition(OperatingSystems.MacOSX)]
```
Comment (dougbu) [inline test/Microsoft.Dnx.TestHost.Tests/TestHostTest.cs]: Make tests changed in this class skipped `[Fact]`s. The conditions only confuse.

File: test/Microsoft.Dnx.TestHost.Tests/project.json [hunk 7]
```diff
@@ -11,14 +11,26 @@
     "Microsoft.Dnx.TestHost.Client.Sources": {
       "type": "build",
       "version": "1.0.0-*"
-    },
-    "xunit.runner.aspnet": "2.0.0-aspnet-*"
```
Comment (dougbu) [inline test/Microsoft.Dnx.TestHost.Tests/project.json]: Why no `xunit`?

File: test/Microsoft.Dnx.TestHost.Tests/TestHostTest.cs [hunk 8]
```diff
@@ -21,11 +21,11 @@ public class TestHostTest
 
         public TestHostTest()
         {
-            var libraryManager = PlatformServices.Default.LibraryManager;
-            _testProject = Path.GetDirectoryName(libraryManager.GetLibrary("Sample.Tests").Path);
+            var testProjectFolder = Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(Directory.GetCurrentDirectory())));
```
Comment (dougbu) [inline test/Microsoft.Dnx.TestHost.Tests/TestHostTest.cs]: `Path.Combine(Directory.GetCurrentDirectory(), "..", "..", "..")` (or so) would be much more readable.

**Comment by dougbu** [comment]
:watch:

File: .travis.yml (hunk 1 above)
Comment (victorhurdugaci) [inline .travis.yml]: Will do as part of the second pass through all the repos.

File: src/Microsoft.AspNet.Testing/project.json (hunk 2 above)
Comment (victorhurdugaci) [inline src/Microsoft.AspNet.Testing/project.json]: It is not a test project

File: src/Microsoft.AspNet.Testing/project.json (hunk 3 above)
Comment (victorhurdugaci) [inline src/Microsoft.AspNet.Testing/project.json]: Not needed. It's not a test project

**Comment by victorhurdugaci** [comment]
Updated.
cc @rynowak

File: test/Microsoft.Dnx.TestHost.Tests/TestHostTest.cs [hunk 9]
```diff
@@ -22,10 +22,10 @@ public class TestHostTest
...
         {
             var libraryManager = PlatformServices.Default.LibraryManager;
-            _testProject = Path.GetDirectoryName(libraryManager.GetLibrary("Sample.Tests").Path);
+            _testProject = "Not set"; // _testProject = Path.GetDirectoryName(libraryManager.GetLibrary("Sample.Tests").Path);
         }
 
-        [ConditionalFact]
+        [Fact(Skip = "TestHost has to be redesigned to support the new project format")]
         [OSSkipCondition(OperatingSystems.Linux)]
         [OSSkipCondition(OperatingSystems.MacOSX)]
```
Comment (dougbu) [inline test/Microsoft.Dnx.TestHost.Tests/TestHostTest.cs]: Sorry, I wasn't clear. Meant you should remove the `[OSSkipCondition]`s as well.

File: test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs [hunk 10]
```diff
@@ -21,12 +21,11 @@ public class ProjectTest
...
         {
             var libraryManager = PlatformServices.Default.LibraryManager;
-
             // Example: "C:\github\testing\test\Sample.Tests"
-            _testProjectPath = Path.GetDirectoryName(libraryManager.GetLibrary(projectName).Path);
+            _testProjectPath = "Not set"; // Path.GetDirectoryName(libraryManager.GetLibrary(projectName).Path);
         }
 
-        [Fact]
+        [Fact(Skip = "TestHost has to be redesigned to support the new project format")]
```
Comment (dougbu) [inline test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs]: Do we have an issue that can be linked here? Does that issue mean `TestHost` is useless at the moment? (We have a fair number of repos using `TestHost` don't we?)

**Comment by dougbu** [comment]
:watch: mostly for new question

File: test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs (hunk 10 above)
Comment (victorhurdugaci) [inline test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs]: This is the test host for Visual Studio. I don't think any of our repos use it except VS. @rynowak ?

Comment (victorhurdugaci) [inline test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs]: Talked to @vijayrkn. The dotnet team is going to provide a test host for the new world

**Comment by victorhurdugaci** [comment]
Need to merge this. It is blocking the progress on dotnet migration...

File: test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs (hunk 10 above)
Comment (dougbu) [inline test/Microsoft.Dnx.TestHost.Tests/ProjectTest.cs]: :ok:

**Comment by dougbu** [comment]
:shipit: w/ the `[OSSkipCondition]` removal.

**Comment by Eilon** [comment]
@victorhurdugaci any update on this?

**Comment by victorhurdugaci** [comment]
Will rebase and merge

**Comment by Eilon** [comment]
Thanks!

**Pull request merged**